examples/geospatial/plot_dissolve_polygons.py

```diff
@@ -0,0 +1,107 @@
+"""
+=====
+Dissolve adjacent polygons
+=====
+
+This example shows how to dissolve adjacent polygons by erasing the boundaries
+between them.
+
+Using a geometric union alone can be quite slow depending on the nature of the
+geometry, so there is sometimes an advantage to only performing a union on those
+geometries that interact spatially, especially if those relationships can be
+determined using a spatial index such as in this example.  Furthermore, it may
+be important to retain a connection back to some of the original attributes of
+the dissolved geometries, whereas performing a union without a grouping step
+first will lose this information.
+
+Dissolve in GeoPandas requires an attribute to dissolve by.  This example uses
+networkx to identify all groups of polygons that intersect each other, and then
+dissolves by that group.
+
+More information on dissolve at:
+https://geopandas.org/aggregation_with_dissolve.html
+
+Note: this example requires either rtree or pygeos optional dependency for
+GeoPandas.
+
+See also
+--------
+plot_polygons.py: provides an example using PySAL to identify contiguous groups
+    of polygons, which could then be unioned together to create the same results
+    as this example.
+"""
+
+import matplotlib.pyplot as plt
+import networkx as nx
+import geopandas
+import pandas
+import numpy
+
+### Read in example data.
+# Input file format could be any format supported by GeoPandas.
+filepath = "nuts1.geojson"
+european_regions = geopandas.read_file(filepath)
+
+### Spatially join regions to themselves using the spatial index.
+# This finds all regions that spatially intersect other regions.
+pairs = pandas.DataFrame(geopandas.sjoin(european_regions, european_regions))[
+    ["index_right"]
+]
```

Comment:
I know this does the job, but seeing it used like this in documentation is not making me happy since it is not very straightforward and not the primary use of sjoin. That is the reason for something like `touches_matrix` proposed in https://github.com/geopandas/geopandas/pull/1674.